core/src/Threads/Kokkos_ThreadsExec.hpp

```diff
@@ -677,15 +728,28 @@ class UniqueToken<Threads, UniqueTokenScope::Global> {
   UniqueToken(execution_space const & = execution_space()) noexcept {}
 
   /// \brief upper bound for acquired values, i.e. 0 <= value < size()
-  inline int size() const noexcept { return Threads::impl_thread_pool_size(); }
+  KOKKOS_INLINE_FUNCTION
+  int size() const noexcept {
```

Comment:
I don't like this pattern of KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST in the Threads backend, but think it's a problem with our macros, not this implementation. Capturing here https://github.com/kokkos/kokkos/issues/3209